src/libs/onboardingSelectors.ts

```diff
@@ -9,8 +10,8 @@ import type ONYXKEYS from '@src/ONYXKEYS';
  * `false` means the user has not completed the NewDot onboarding flow
  */
 function hasCompletedGuidedSetupFlowSelector(onboarding: OnyxValue<typeof ONYXKEYS.NVP_ONBOARDING>): boolean | undefined {
-    // Onboarding is an array for old accounts and accounts created from OldDot
-    if (Array.isArray(onboarding)) {
+    // Onboarding is an array or an empty object for old accounts and accounts created from OldDot
+    if (Array.isArray(onboarding) || isEmptyObject(onboarding)) {
```

Comment:
I think this would not work as expected - onboarding NVP is empty for new users and we would return true from this function and never display onboarding setup thinking the user has already completed the setup 🤔 ?

Reply:
Did you create a new account? I logged into one of my old accounts and the NVP was returned as empty object.

Reply:
> onboarding NVP is empty for new users and we would return true from this function and never display onboarding setup thinking the user has already completed the setup 🤔 ?

This is only for new users who sign up in OldDot or they are on domain controlled account., they wont get the nvp set up but we also do not want to show them the onboarding as it might not be relevant for them. We only want to show the onboarding to users who signed up through newDot and in that case, they will have the necessary property set in the NVP

Reply:
Sounds good! Thanks for clarifying 👍